dbrequests/mysql/connection.py

```diff
@@ -0,0 +1,134 @@
+"""Implements the send_data method using load data local infile.
+
+This is mysql and mariadb compliant.
+"""
+from tempfile import NamedTemporaryFile as TmpFile
+from dbrequests import Connection as SuperConnection
+from contextlib import contextmanager
+
+
+class Connection(SuperConnection):
+    """A Database connection."""
+
+    def send_data(self, df, table, mode='insert', **params):
```

Comment:
Not sure here. `dbrequests.Connection` has a `send_data` method and this class overrides it. Did you really mean `send_data` or something else?

Reply:
I just think that the Connection class (and especially other child classes) would benefit from having send_data written in the way you did it here in the mysql subclass.

Reply:
I updated the method in the super class and refactored the whole thing, so that a child would only have to override the modes. send_data will be the same for all derived classes. Hope that is what you had in mind.

I also added a deprecation warning to the update method in the superclass. I think this PR is already too big, but we should think about removing the update/replace mode from the super class. Essentially what you had in mind with the original #11 ticket.

Reply:
I had to change quite a bit. The reason was that we have to be open for any set of modes in a child class. I accomplished that with `getattr`. I found that on the internet and saw it also somewhere else. Not sure if this is pythonic, but it does the trick. I struggle to see how one should document additional modes. The only place would be in a child class in the docstring. For now I can live with it.

Reply:
Looks good!

Reply:
I agree on deprecation of these modes. There is a small typo in the deprecation warnings ("the mode is depreacted").